core/heap.c

```diff
@@ -157,7 +159,7 @@ DECLARE_NEVERPROT_VAR(static bool out_of_vmheap_once, false);
 #endif
 
 /* variable-length: we steal one int for the size */
-#define HEADER_SIZE (sizeof(size_t))
+#define HEADER_SIZE HEAP_ALIGNMENT
```

Comment:
If headers were used for every allocation, doubling the header size would probably be too expensive to bear.  Since only allocs that don't fit in the buckets use headers: this is probably ok; client heap allocations probably bear the biggest hit.

Please update the comment ("steal one int": this is more than an int size) and elaborate to explain the wasted space is not expected to be significant for DR's own use.